### DISCUSSION TITLE
关于同一台服务器，使用本地代理、搭建xray、搭建alist-proxy、rclone四种方式下载速度的提问

### DISCUSSION THREAD

**Discussion by Minerchu** [AlistGo/alist#4988] [Q&A]
因为使用的网盘是onedrive和Googledriver
一个直连速度很慢，一个是直连无法访问。
所以日常使用中，都在用代理下载。

四种方式都有试过速度、每个速度都不一样

1、本机代理
使用服务器本身alist转发，下载google drive稳定在50~100kb/s

2、使用alist服务器搭建xray，然后alist 302 到源链接，xray开全局下载
速度在10m/s左右晃

3、使用本机搭建alist-proxy，在另一台机器搭建的alist下载，速度在5-6m/s

4、本机使用rclone，下载资源，服务器瞬时下行能400M/S左右，上行20M/S

四种方式最实用的其实是1，最简单，最省事，但是也是最慢的。
rclone很快，但是基本上离开电脑就无法使用，所以pass
2的使用场景有限，而且还要单独套一层代理，有点麻烦
3也很方便，但是速度不咋地，下载4k资源有时候会突然卡住。

所以大家都是什么方式使用下载代理的功能呢？

怎么才能让下载的传输效率提高一点

**Comment by ypq123456789** [comment]
你都用服务器了，为啥不在这个服务器上搭建alist并开启本地代理呢，就不用alist-proxy了呀

**Reply by Minerchu** [reply]
是这样，没错啦😅
我用的服务器是10gpbs的，然后在alist里添加gd，速度不过10兆一秒🥹不管是走本机还是alistproxy都一样
但是自己rclone挂载的gd盘，可以跑300-400多兆一秒
不知道哪里出问题

**Reply by ypq123456789** [reply]
我用cf的workers也经常出错，之前让claude2看过alist-proxy的代码，说是可以进一步优化，缺少了缓存部分，我让他写了，但没敢替换。索性就用服务器本地代理，快且稳定，唯一的坏处是要浪费流量

**Reply by Minerchu** [reply]
我无限流量，这个倒是还好……主要是速度跑不上来好难受😂

**Comment by Arc-2023** [comment]
emm，
## 是因为Cloudflare的问题：给workers分配的IP貌似是动态的，这些IP线路质量参差不齐
所以有的时候proxy用起来很快，能跑满千兆，有的时候用aria2 512个线程都跑不到2MB，
目前的解决方案就是用国外服务器代理下载就好，**用xray翻出去下载或者国外服务器部署Alist并开启本地代理**
当然你也可以用xray和workers搞链式代理，不过这样适用于国内的服务器和国内的云盘了
至于服务器就可以选择racknerd这种的大带宽，低价格的来用了

**Reply by Minerchu** [reply]
昂我的意思是，同一台服务器，搭xray跑网速可以接近300-400m的速率，搭建alist后本地代理下载Googledriver才10m/s不到，大多数时候是2-3m/s

**Reply by coldnight9** [reply]
workers也可以优选多个ip使用，我可以跑满千兆

**Reply by Minerchu** [reply]
啊移动联通还是电信啊 ，我cf优选是cname到stock.hostmonit.com，你的可以分享一下吗🥹🥹

**Comment by ypq123456789** [comment]
能细说一下怎么操作吗哥？感谢🙏



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年8月16日(周三) 凌晨2:45
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [alist-org/alist] 关于同一台服务器，使用本地代理、搭建xray、搭建alist-proxy、rclone四种方式下载速度的提问 (Discussion #4988)




 
workers也可以优选多个ip使用，我可以跑满千兆

**Comment by KnIfER** [comment]
不完整下载的话，随机读取性能如何？
